nameable.rb

```diff
@@ -0,0 +1,31 @@
+# create nameable class
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    return unless @nameable.correct_name.length > 10
+
+    super[0...10]
+  end
+end
```

Comment:
[OPTIONAL]

- I will strongly recommend you separate your classes to there separate files so that your code can be modular and clean

Reply:
Since this is optional, is it okay if I revisit it later?

Reply:
@shaqdeff , please, I will encourage you to implement it now. Please prioritize writing clean code

Reply:
@vickymarz Alright. Again, thanks a lot. I appreciate your candor and professionalism.